### ISSUE TITLE
Realtime demo

### ISSUE THREAD

**Issue by jinglan0829** [zju3dv/clean-pvnet#6]
感谢作者的分享！
请问作者如何用训练出来的weight 来实时(Realtime)检测object的pose呢？
例如像主页里发布的Real-time-demo视频一样，可以在编码上稍微指点一下实时检测物体位姿的部分吗？
期待作者的回复，谢谢！

**Comment by pengsida** [comment]
我们用detector先检测图片中的物体，然后将物体crop，输入pvnet，检测pose。

**Comment by jinglan0829** [comment]
kpt_2d = output['kpt_2d'][0].detach().cpu().numpy()
img_id = int(batch['img_id'][0])
anno = self.coco.loadAnns(self.coco.getAnnIds(imgIds=img_id))[0]
 kpt_3d = np.concatenate([anno['fps_3d'], [anno['center_3d']]], axis=0)
 K = np.array(anno['K'])
pose_gt = np.array(anno['pose'])
pose_pred = pvnet_pose_utils.pnp(kpt_3d, kpt_2d, K)
但计算物体位姿部分, kpt_3d我发现是用数据集里的物体mask，pose来计算（anno)，可是实时检测到的物体并不是数据集里的，这个部分该怎么处理呢

**Comment by pengsida** [comment]
我们实时demo的物体是数据集里的。如果模型不已知，就没有6d pose的概念。

**Comment by jinglan0829** [comment]
实时demo的物体是数据集里的，但是实时demo得到的物体的图片是随机的，而anno这个部分是是根据数据集里的顺序排序的

**Comment by pengsida** [comment]
kpt_3d是固定的

**Comment by jinglan0829** [comment]
好的，谢谢！

**Comment by kevin860213** [comment]
那請問你在real times 時會得到物品的crop 這包含了mask嗎?謝謝

**Comment by pengsida** [comment]
不包括，mask是pvnet预测的。
